apps/keyboard/js/keyboard/target_handlers_manager.js

```diff
@@ -107,7 +107,7 @@ TargetHandlersManager.prototype._callTargetAction = function(action,
   }
 
   this.promiseQueue = this.promiseQueue.then(function() {
-    handler[action](press);
+    return handler[action](press);
```

Comment:
Yes, just check this again on Flame with 319MB and the performance should be fine.